lib/sorcery/model/submodules/brute_force_protection.rb

```diff
@@ -63,13 +66,14 @@ def define_brute_force_protection_fields
         module InstanceMethods
           # Called by the controller to increment the failed logins counter.
           # Calls 'login_lock!' if login retries limit was reached.
-          def register_failed_login!
+          def register_failed_login!(password)
```

Comment:
Is the password parameter necessary here? The method simply registers a failed login attempt, its behavior should not depend on the provided password. 

Can we just reverse the order of increment call and `return unless login_unlocked?`. Will that have the desired effect?

Reply:
As mentioned, passing the password to this callback is unnecessary at best, and malicious at worst.